File: user-service/src/main/java/socialapp/userservice/repository/UserRepository.java
package socialapp.userservice.repository;

import org.springframework.stereotype.Repository;
import socialapp.userservice.model.entity.User;
import socialapp.userservice.repository.common.CommonRepository;

@Repository
public interface UserRepository extends CommonRepository<User, Long> {

    Boolean existsByEmail(String email);

    @Override
    default Class<?> entityClass(){
        return User.class;
    };

}

File: user-service/src/main/kotlin/socialapp/ktuserservice/repository/UserRepository.kt
package socialapp.ktuserservice.repository

import org.springframework.stereotype.Repository
import socialapp.ktuserservice.model.entity.User
import socialapp.ktuserservice.repository.common.CommonRepository

@Repository
interface UserRepository : CommonRepository<User, Long> {

    fun existsByEmail(email: String): Boolean

    fun findByEmail(email: String): User?

    override fun entityClass(): Class<*> = User::class.java
}

